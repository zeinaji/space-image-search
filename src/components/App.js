import React, { useState } from "react";
import "../styles/App.css";
import Search from "./Search";
import SearchResults from "./SearchResults";

function App() {
  const [searchResults, setSearchResults] = useState({});

  return (
    <div className="App">
      <img
        className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="logo"
      />
      <Search setSearchResults={setSearchResults} />
      <SearchResults image="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80" />
    </div>
  );
}

export default App;
