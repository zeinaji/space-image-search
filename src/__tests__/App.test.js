import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  render(<App />);
  xit("it renders correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("it renders the logo", () => {
    expect(screen.getByAltText("logo")).toHaveClass("nasa-logo");
  });
});
