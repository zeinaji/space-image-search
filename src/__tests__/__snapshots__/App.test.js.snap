// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
    >
      <form
        class="form-wrapper"
      >
        <input
          class="search-input"
          type="text"
        />
        <button
          class="submit"
          type="submit"
        >
          Go!
        </button>
      </form>
    </div>
    <img
      alt="search-result"
      src="https://images.unsplash.com/photo-1522030299830-16b8d3d049fe?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
    />
  </div>
</DocumentFragment>
`;
